internal/integration/integration_test.go

```diff
@@ -0,0 +1,392 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integration
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"sort"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/google/exposure-notifications-server/internal/database"
+	exportapi "github.com/google/exposure-notifications-server/internal/export"
+	exportdb "github.com/google/exposure-notifications-server/internal/export/database"
+	"github.com/google/exposure-notifications-server/internal/pb/export"
+	publishdb "github.com/google/exposure-notifications-server/internal/publish/database"
+	publishmodel "github.com/google/exposure-notifications-server/internal/publish/model"
+	"github.com/google/exposure-notifications-server/internal/storage"
+	"github.com/google/exposure-notifications-server/internal/util"
+	verifyapi "github.com/google/exposure-notifications-server/pkg/api/v1alpha1"
+	"github.com/google/exposure-notifications-server/pkg/base64util"
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	pgx "github.com/jackc/pgx/v4"
+	"github.com/sethvargo/go-retry"
+	"google.golang.org/protobuf/proto"
+)
+
+func TestIntegration(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+	env, client, db := NewTestServer(t, 2*time.Second)
+
+	// Set query criteria (used throughout)
+	criteria := publishdb.IterateExposuresCriteria{
+		OnlyLocalProvenance: false,
```

Comment:
can add a TODO / file an issue- this only gets non-revised keys.
we'll also want to add the ability to revise keys in the test and iterate over the revised keys